### PR TITLE
CI: Build ad-hoc deploy previews (deployment with efs instead of sts)

### PR DESCRIPTION
**Pull request by sum35h**
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.
>
#### PR fixes following issue(s)
Fixes # (issue number)
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: scripts/build_dp_from_branch.sh [hunk 1]
```diff
@@ -34,8 +35,8 @@ export SECRET="$SUB_DOMAIN_NAME"
 export DBNAME="$SUB_DOMAIN_NAME"
 export DOMAINNAME="$SUB_DOMAIN_NAME".dp.appsmith.com
 export HELMCHART="appsmith"
-export HELMCHART_URL="http://helm.appsmith.com"
-export HELMCHART_VERSION="2.0.2"
+export HELMCHART_URL="http://helm-ee.appsmith.com"
+export HELMCHART_VERSION="3.0.7"
```
Comment (sharat87) [inline scripts/build_dp_from_branch.sh:39]: Is this change intentional? 🤔

File: scripts/build_dp_from_branch.sh [hunk 2]
```diff
@@ -60,17 +82,29 @@ kubectl create secret docker-registry $SECRET \
   --docker-password=$DOCKER_HUB_ACCESS_TOKEN -n $NAMESPACE
 
 echo "Add appsmith-ce to helm repo"
-AWS_REGION=ap-south-1 helm repo add $HELMCHART $HELMCHART_URL
+AWS_REGION=ap-south-1 helm repo add appsmith-ee $HELMCHART_URL;
+helm repo update;
+helm plugin install https://github.com/helm/helm-mapkubeapis -n $NAMESPACE
+helm plugin ls
+helm mapkubeapis $CHARTNAME -n $NAMESPACE
```
Comment (sharat87) [inline scripts/build_dp_from_branch.sh:89]: Can we add a comment on what this plugin is about please?

File: scripts/build_dp_from_branch.sh (hunk 1 above)
Comment (sum35h) [reply scripts/build_dp_from_branch.sh:39]: Yes, it is. Looks like we're using the ee helm chart for all dp's.

File: scripts/build_dp_from_branch.sh (hunk 2 above)
Comment (sum35h) [reply scripts/build_dp_from_branch.sh:89]: This plugin prevents old deployments from breaking when the k8s api are updated.
This was applied earlier by Goutham when we upgraded to v1.25 since the dp's started to break.
https://github.com/helm/helm-mapkubeapis

File: scripts/build_dp_from_branch.sh [hunk 3]
```diff
@@ -47,10 +48,33 @@ kubectl get pods
 
 if [[ -n "${RECREATE-}" ]]
 then
-  kubectl delete ns $NAMESPACE || true
   mongosh "mongodb+srv://$DB_USERNAME:$DB_PASSWORD@$DB_URL/$DBNAME?retryWrites=true&minPoolSize=1&maxPoolSize=10&maxIdleTimeMS=900000&authSource=admin" --eval 'db.dropDatabase()'
+  pod_name=$(kubectl get pods -n $NAMESPACE -o json | jq '.items[0].metadata.name' | tr -d '"')
```
Comment (sum35h) [inline scripts/build_dp_from_branch.sh:52]: Copied this over from the other dp script. This is not yet used in the ad-hoc dp's.
